doc/dev/kdb-add-new-command.md

```diff
@@ -0,0 +1,170 @@
+# Implement new commands in C
+
+Here we'll go over the basic requirements to implement a new command in C and integrate it in to the `kdb` CLI framework.
+
+## 1. Files
+
+First we need a header as well as a c file for the new command in `src/tools/kdb`.
+
+### Header
+
+The `<name>.h` has to look something like this:
+
+```c
+/**
+ * @file
+ *
+ * @brief Header for ... command
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_KDB_..._H
+#define ELEKTRA_KDB_..._H
+
+#include <kdb.h>
+
+/**
+ * Adds options specification of complete command to @spec
+ *
+ * @param spec the base spec where the commands spec should be added
+ */
+void add...Spec (KeySet * spec);
+
+/**
+ * Executes the ... command
+ *
+ * @param options cli options and arguments as specified in addCompleteSpec()
+ * @param errorKey key where errors and warnings should be saved
+ *
+ * @retval 0 ... command ran without errors
+ * @retval 1 errors occurred, keyGetMeta (errorKey, "error/reason") for info
+ *
+ */
+int exec... (KeySet * options, Key * errorKey);
+
+#endif // ELEKTRA_KDB_..._H
+```
+
+### `C` file
+
+The `<name>.c` to implement the two functions already described in the header file, in addition to those two functions `COMMAND_NAME`, `GET_OPTION_KEY` and `GET_OPTION` have to be defined as follows:
+
+```c
+#define COMMAND_NAME "..." // your command name
+
+#define GET_OPTION_KEY(options, name) GET_OPT_KEY (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)
+#define GET_OPTION(options, name) GET_OPT (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)
+```
+
+It should also `#include <colors.h>` and `#include <command.h>`, those contain all the needed helper macros and function.
+
+#### Specification
+
+`void add...Spec (KeySet * spec)` will contain the specification (for `elektraGetOpts`) of the new command.
+It has to append it to the passed `KeySet * spec`, the framework will call this function to register its specification.
+The function could look something like this:
+
+```c
+void addGetSpec (KeySet * spec)
+{
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME), KEY_META, "description", "Get the value of an individual key.",
+				   KEY_META, "command", COMMAND_NAME, KEY_END));
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/all", KEY_META, "description", "Consider all of the keys", KEY_META,
+				   "opt", "a", KEY_META, "opt/long", "all", KEY_META, "opt/arg", "none", KEY_END));
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/name", KEY_META, "description", "The name of the key", KEY_META,
+				   "args", "indexed", KEY_META, "args/index", "0", KEY_END));
+
+	ADD_BASIC_OPTIONS (spec, COMMAND_SPEC_KEY (COMMAND_NAME))
+}
+```
+
+The `ADD_BASIC_OPTIONS` at the end adds options that are common across all commands to its spec, this includes things like `verbose: -v` or `debug: -d`.
+It if your command should support these, it makes sense to use the macro.
```

Comment:
```suggestion
It if your command should support these, use the macro.
```

Are there exceptions where they should not be supported? If not remove the sentence:

```suggestion
```